### PR TITLE
Bunch of random fixes

### PR DESCRIPTION
**Pull request by krzkaczor**
Mostly stuff that I noticed while running backend locally:
* fixed logger names for the activity
* untangled StarkexCounter code by introducing `uncertaintyBuffer` directly to SequenceProcessor
* Fix bug in StarkexCountRepository
* Improved start_db.sh script

### CODE REVIEW

File: packages/backend/src/core/SequenceProcessor.ts [hunk 1]
```diff
@@ -117,9 +121,14 @@ export class SequenceProcessor extends EventEmitter {
   private async process(): Promise<void> {
```
Comment (antooni) [inline packages/backend/src/core/SequenceProcessor.ts:121]: This function is pretty big and hard to read without greater context. Definietely some comment at the top would be useful, or to be more clean splitting it into smaller pieces to be more self-explainatory. But this one should not be in the scope of this PR.

File: packages/backend/src/core/SequenceProcessor.ts [hunk 2]
```diff
@@ -117,9 +121,14 @@ export class SequenceProcessor extends EventEmitter {
...
     this.logger.debug('Processing started')
 
+    let firstRun = true
```
Comment (antooni) [inline packages/backend/src/core/SequenceProcessor.ts:124]: This variable looks a bit ugly at the first sight, but after reading this code I must say this solution is very clean.

File: packages/backend/src/core/SequenceProcessor.ts [hunk 3]
```diff
@@ -47,6 +49,7 @@ interface State {
 export class SequenceProcessor extends EventEmitter {
```
Comment (antooni) [inline packages/backend/src/core/SequenceProcessor.ts:49]: contructor of this class is a bit unexpected for me, I thought that in our codebase we treat constructor as only property setting function. In this case inside the constructor `process()` is added to queque. IMO we should try to have the same behaviour repo-wise and change this flow to sth like: 
```
create object
start()
```

This should not be in the scope of this PR

**Review by antooni — APPROVED** [review]

**Review by michalsidzej — APPROVED** [review]
LGTM, but maybe @sz-piotr has something to add

File: packages/backend/src/peripherals/database/shared/types.ts [hunk 4]
```diff
@@ -94,3 +94,7 @@ declare module 'knex/types/tables' {
     'activity.daily_count_view': DailyTransactionCountRow
   }
 }
+
+// Some aggregations return not empty row with null values. Use this type to explicitly type them.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type OptionalDict<T = any> = Record<string, T | null>
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/shared/types.ts:100]: Maybe NullableDict instead?

File: packages/backend/src/core/SequenceProcessor.ts [hunk 5]
```diff
@@ -117,9 +121,14 @@ export class SequenceProcessor extends EventEmitter {
   private async process(): Promise<void> {
     this.logger.debug('Processing started')
 
+    let firstRun = true
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     processing: while (true) {
-      const lastProcessed = this.state?.lastProcessed
+      let lastProcessed = this.state?.lastProcessed
+      // we need to adjust starting block if its first run and uncertaintyBuffer is set
+      if (firstRun && lastProcessed && this.uncertaintyBuffer > 0) {
```
Comment (sz-piotr) [inline packages/backend/src/core/SequenceProcessor.ts:129]: This implicitly relies on heroku's behavior of restarting the process every day. Maybe if uncertainty buffer is set we should always set `lastProcessed = actualLastProcessed - uncertaintyBuffer`? Or only if we're on the tip?

Comment (krzkaczor) [reply packages/backend/src/core/SequenceProcessor.ts:129]: I am not sure if I understand, `process()` gets called on hourly basis so it should work fine. Please ping me if I am wrong but I am gonna merge it now.

**Pull request merged**